report-ng/app/src/components/timings/test-timings/test-timings.ts

```diff
@@ -0,0 +1,372 @@
+/*
+ * Testerra
+ *
+ * (C) 2023, Telekom MMS GmbH, Deutsche Telekom AG
+ *
+ * Deutsche Telekom AG and all other contributors /
+ * copyright owners license this file to you under the Apache
+ * License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {autoinject, bindable, observable} from 'aurelia-framework';
+import {NavigationInstruction, RouteConfig} from "aurelia-router";
+import {AbstractViewModel} from "../../abstract-view-model";
+import * as echarts from 'echarts';
+import {EChartsOption} from 'echarts';
+import "./test-timings.scss";
+import {ExecutionStatistics} from "services/statistic-models";
+import {MethodDetails, StatisticsGenerator} from "services/statistics-generator";
+import moment from "moment";
+import {data} from "../../../services/report-model";
+import {StatusConverter} from "../../../services/status-converter";
+import MethodType = data.MethodType;
+import IMethodContext = data.MethodContext;
+
+@autoinject()
+export class TestTimings extends AbstractViewModel {
+    private _chart: echarts.ECharts;
+    private _executionStatistics: ExecutionStatistics
+    private _option: EChartsOption;
+    private _hasEnded = false;
+    private _methodDetails: MethodDetails[];
+    private _labels: string[];
+    private _sectionValues: number[];
+    private _data: number[];
+    private _bars: IDurationBar[];
+    private _loading = false;
+    private _searchRegexp: RegExp;
+    private _inputValue = '';
+    private _methodId;
```

Comment:
These member could also use types, i.e. `_methodId` is used both as a string and a number.